thunderfish/fakefish.py

```diff
@@ -0,0 +1,173 @@
+    data += 0.05 * np.random.randn(len(data))
+
+    return data
+
+
+def generate_pulsefish(freq, samplerate, time_len=20., noise_fac=0.1, wave_cut=0.6,
+                       pk_len=0.001, pk_std=0.1, peak_fac=2.,
+                       tr_len=0.001, tr_std=0.12, tr_fac=1.2):
+    """
+
+    :param freq: (float). Frequency of the fish in Hz.
+    :param samplerate: (float). Sampling Rate in Hz
+    :param time_len: (float). Length of the recording in sec.
+    :param noise_fac: (float). Factor by which random gaussian distributed noise is inserted.
+    :param wave_cut: (float). Ideally between 0.5 and 1. marks the percentage at which the distribution is cut.
+    :param pk_len: (float). length of the positive part of the pulse.
+    :param pk_std: (float). std of the positive part of the pulse.
+    :param peak_fac: (float). Factor for regulating the positive part of the pulse.
+    :param tr_len: (float). length of the negative part of the pulse.
+    :param tr_std: (float). std of the negative part of the pulse.
+    :param tr_fac: (float). Factor for regulating the negative part of the pulse.
+
+    :return: (array). Data with pulses at the given frequency.
+    """
+
+    def gaussian(x, mu, si):
+        """
+        Standard Gaussian function taken from Wikipedia.
+
+        :param x: (array-like). An arange of numbers.
+        :param mu: (float). The Mean, which in this case should be preferably 0.
+        :param si: (float). The Standard Deviation.
+        :return:
+        """
+        return (1./np.sqrt(2*si**2*np.pi)) * np.exp(-(x-mu)**2/(2*si**2))
+
+    def get_slope(peak, trough, length_percent=0.05):
+        """Calculates the slope that connects the positive part of the pulse with the negative one.
+
+        :param peak: (array-like). the positive values of the pulse
+        :param trough: (array-like). the negative values of the pulse
+        :param length_percent: (float). How long (in percent) the slope should be compared to the whole pulse length
+        :return: (np.array). returns the slope values
+        """
+        slope_len = (len(peak) + len(trough)) * length_percent  # length of slope respective to length of entire pulse
+        s = np.linspace(peak[-1], trough[0], slope_len)
+        if len(s) < 3:
+            s = np.array([(peak[-1] + trough[0])/2.])
+        return s
+
+    def insert_pulses(freq, pulse, time_len, noise_fac):
+        """Insert pulses into noisy baseline at a given frequency
+
+        :param freq: s.a.
+        :param pulse: s.a.
+        :param time_len: s.a.
+        :param noise_fac: s.a.
+        :return: (array). Data with pulses at the given frequency.
+        """
+        time = np.arange(0, time_len, 1. / samplerate)
+
+        noise = (np.random.randn(len(time)) * noise_fac)
+        dat = np.zeros(len(time)) * noise
+
+        bl_idx = np.arange(len(dat))
+        pulse_start = bl_idx[bl_idx % (samplerate/freq) == 0.]
+
+        for s in pulse_start:
```

Comment:
period = int(samplerate/freq)
for s in range(period/2, len(dat)-period/2, period):

with that lines 87 and 88 can be removed.